File: factors.c
#include <stdio.h>
#include <math.h>

int main(int argc, char *argv[]) {
    FILE *file;
    int number, i;
    if (argc != 2) {
        printf("Usage: %s <file>\n", argv[0]);
        return 1;
    }
    file = fopen(argv[1], "r");
    if (!file) {
        printf("Error: Cannot open file %s\n", argv[1]);
        return 1;
    }
    while (fscanf(file, "%d", &number) == 1) {
        printf("%d=", number);
        for (i = 2; i <= sqrt(number); i++) {
            if (number % i == 0) {
                printf("%d*%d\n", i, number/i);
                break;
            }
        }
        if (i > sqrt(number)) {
            printf("%d\n", number);
        }
    }
    fclose(file);
    return 0;
}

File: factors.py
import sys
import math

def main(filename):
    with open(filename, 'r') as file:
        for line in file:
            number = int(line.strip())
            print(f"{number}=", end="")
            for i in range(2, int(math.sqrt(number)) + 1):
                if number % i == 0:
                    print(f"{i}*{number//i}")
                    break
            else:
                print(f"{number}")

if __name__ == '__main__':
    if len(sys.argv) != 2:
        print("Usage: python factors.py <file>")
        sys.exit(1)
    main(sys.argv[1])
